packages/firestore/src/local/shared_client_state_syncer.ts

```diff
@@ -14,10 +14,26 @@
  * limitations under the License.
  */
 
-import { BatchId, MutationBatchState } from '../core/types';
+import { BatchId, MutationBatchState, TargetId } from '../core/types';
 import { FirestoreError } from '../util/error';
 import { ClientId } from './shared_client_state';
 
+/**
+ * The different states of a watch target.
+ *
+ * When a secondary tab inserts a new target, it is marked 'pending'. The
+ * primary tab then transitions the target to 'incomplete' once Watch has
+ * delivered the first snapshot. Once a Watch target is current, it transitions
+ * to 'current'.
+ *
+ * Targets that get rejected from the backend are marked 'rejected'.
+ */
+export type QueryTargetState =
+  | 'pending'
+  | 'incomplete'
```

Comment:
What's the advantage of 'incomplete' over 'not-current' (which avoids inventing a new term we don't use in the rest of the client)?

Reply:
It's not as clumsy as 'not-current', but I don't feel strongly. I changed it everywhere (hopefully!) to not-current.